forc-pkg/src/source/git/mod.rs

```diff
@@ -228,6 +225,19 @@ impl source::DepPath for Pinned {
     }
 }
 
+impl AsRef<gix_url::Url> for Url {
+    fn as_ref(&self) -> &gix_url::Url {
+        &self.url
+    }
+}
```

Comment:
Do we use this implementation anywhere? If not, perhaps we can leave this out? Might be good to avoid exposing the inner `gix_url::Url` if we don't need to.

Reply:
I am using it for the `display` implementation but I can remove it as well, I originally added it since `gix_url::Url` provides nice utilities such as [this](https://docs.rs/gix-url/latest/gix_url/struct.Url.html#method.user) one, which might be useful in the future for more edge-case ssh authentication purposes (especially we might end-up checking the url for a username before trying the agent, I haven't added it as I need to experiment with it)

Reply:
Yeah nice, I can see that maybe being useful. I would probably still omit this until we need it, but I'm happy either way

Reply:
All right, I am converting to draft and will remove it first thing. Thanks for the comment 🙌